### ISSUE TITLE
Add transform method

### ISSUE THREAD

**Issue by alexesDev** [planttheidea/unchanged#24]
like:
```js
const transform = curry((path, fn, object) => set(path, fn(get(path, object)), object));

setState(transform('model.counter', x => x + 1, this.state));
// or
import { withProps } with 'recompact';
import moment from 'moment';

const enhance = withProps(transform('data.takenAt', moment))
```

**Comment by planttheidea** [comment]
Sorry I just saw this ... Is this no longer a request?

**Comment by alexesDev** [comment]
This a very small change, but I will be glad to see it in the lib.
I few examples from my project:
```js
withProps(transform('data.meta', JSON.parse))
withProps(transform('data.items.nodes', items => items.map(transform('meta', JSON.parse))))
```

**Comment by planttheidea** [comment]
Sure thing, should not be too crazy to do. I'll whip it up this weekend.

**Comment by planttheidea** [comment]
Alright, I've published the requested `transform` method in `1.4.0`. I gave it a little something extra as well, the ability to provide extra parameters:

```javascript
const fn = (currentValue, extra) => `${extra}_${currentValue}`;

transform('foo.bar', fn, object, 'prefix');
```

These are optional of course (your original use-case still works as expected), but may provide additional benefits for certain cases.

If you have any issues, let me know!

**Comment by alexesDev** [comment]
@planttheidea Good solution with extraParams. Thanks!

**Comment by planttheidea** [comment]
@alexesDev just because they name things that way doesn't mean its the best naming convention. The idea that you are "`adjust`ing" the object is equally accurate as the idea you are "`transform`ing" the object, and the most important thing is that the name of the method makes sense and communicates to the user what is happening. I didn't run with `transform` because you chose it ... I ran with it because I agreed with it. If I named things just because `ramda` named them a specific way, `set` would be named `assocPath` (which I consider to be kind of a terrible name).

I may rename it in the future, though, based on an idea that I've had for a while. I've given some though to this exact use-case before, and it would be a larger change where each of the existing functions would have a companion `*With` function that used callback-based handlers instead of static ones. `getWith`, `removeWith`, `callWith`, and what is now `transform` would be `setWith`. I'm still noodling over what the use case for a callback-based version of `get` would work like, but `remove` could see some benefit (for example, only perform the remove if non-`null` is returned from the callback). A larger change like that, where some potentially larger use-case improvements would result, would be validation for me to do a major breaking change. But that is down the road.

For now, rest easy ... the name is just fine. :smile:

**Comment by alexesDev** [comment]
@planttheidea thanks for the explanation. You're a cool dude :smile:

**Comment by jetpacmonkey** [comment]
Pretty sure this is called `update` in the lodash version. Might be worth calling out in the README?

**Comment by planttheidea** [comment]
@alexesDev 	@jetpacmonkey 	

Hi all! I know it's been a while on this thread, but life's been crazy. Only recently have I been able to work on the refactor I mentioned above, and I'm getting very close. A quick summary of the updates:

- Everything written in TypeScript (hopefully better types, although I'm kind of new to TS)
- Added `is` method, which checks whether the value at the path provided is equal to the value passed based on SVZ
- Every method has a corresponding `*With` method, which receives a callback function as the first parameter, and based on the method does transformation before execution
	* For most methods, it transforms the value that is retrieved / set / merged / etc.
	* For `removeWith`, it acts as validator; falsy returns will not remove the value from the object, truthy returns will
	* `transform` has been renamed to `setWith` as planned (hence my inclusion of this note in this thread)

There are some other things related to the library itself, like gets and sets are faster, and its almost exactly the same size despite all the added functionality, but those are tertiary to the new stuff.

I've finished all the tests, I'm just updating the documentation. Once that is in place ... would either of you be interested in trying the beta? I don't do betas too frequently except on my more popular libraries, and most of the changes are additive or under the hood, the only real breakage is the signature for `transform` (now `setWith`). If you want to try before you buy, let me know, otherwise if there is radio silence I'll probably do a release sometime this weekend.

Happy new year!
